jnosql-mongodb/src/main/java/org/eclipse/jnosql/databases/mongodb/communication/MongoDBDocumentManager.java

```diff
@@ -202,17 +203,34 @@ public long delete(String collectionName, Bson filter) {
      *
      * @param collectionName the collection name
      * @param pipeline the aggregation pipeline
-     * @return the number of documents deleted.
+     * @return the stream of BSON Documents
      * @throws NullPointerException when filter or collectionName is null
      */
-    public Stream<Map<String, BsonValue>> aggregate(String collectionName, List<Bson> pipeline) {
+    public Stream<Map<String, BsonValue>> aggregate(String collectionName, Bson[] pipeline) {
```

Comment:
Why not use `varargs` for the pipeline parameter instead of using an array one? 

Something like that: 

```suggestion
    public Stream<Map<String, BsonValue>> aggregate(String collectionName, Bson... pipeline) {
```


I think `vargargs` offer flexibility to developer because if makes it unnecessary to wrap the Bson instances in an array.

Reply:
Because Java 1.0.2 didn't have it... :-)
That's just another way to pass an array...